### PR TITLE
some stylistic improvements to toTypeFunction()

### PR DESCRIPTION
**Pull request by WalterBright**
https://github.com/dlang/dmd/pull/7749 got merged before I reviewed it, so my review is coming in the form of a touch-up PR.

1. Please don't invent a new formatting style for function Ddoc comments.

2. I find default arguments make code harder to understand. Besides, this is an almost ideal use case for an `out` parameter. `out` parameters eliminate the null pointer issue, are more checkably memory safe, and the code is simpler.

3. Reorganized the code a bit to eliminate variable `tf` and reduce the scope of `s`.

cc @UplinkCoder @JinShil

### CODE REVIEW

File: src/dmd/traits.d [hunk 1]
```diff
@@ -445,49 +445,46 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
...
         {
             if (t.ty == Type.Kind.function_)
-                tf = cast(TypeFunction)t;
+                return cast(TypeFunction)t;
             else if (t.ty == Type.Kind.delegate_)
-                tf = cast(TypeFunction)t.nextOf();
+                return cast(TypeFunction)t.nextOf();
             else if (t.ty == Type.Kind.pointer && t.nextOf().ty == Type.Kind.function_)
-                tf = cast(TypeFunction)t.nextOf();
+                return cast(TypeFunction)t.nextOf();
```
Comment (andralex) [inline src/dmd/traits.d]: two redundant elses, but I'll let them be

**Review by andralex — APPROVED** [review]

**Comment by JinShil** [comment]
@WalterBright, I'm sorry if you felt this was merged prematurely.  But please note that the PR is actually almost 9 months old, and I assumed you were OK with it due to [this comment from you](https://github.com/dlang/dmd/pull/6835#issuecomment-304488000) in the original PR.

> LGTM provided the function is documented

**Comment by WalterBright** [comment]
I reserve the right to learn how to do things better over time :-)

File: src/dmd/traits.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/traits.d]: You're right, I missed that.

**Comment by stefan-koch-sociomantic** [comment]
The comment was written a way I would write code: with indentation. I don't mind the change too much.
The default parameter was used in order to not force the caller to pass in a variable when it's not needed.
Since it was only needed in one case.

also after your changes toTypeFunction will be unable to go through the inliner; which I also considered when writing it.

**Pull request merged**